gdal/swig/python/scripts/gdal_calc.py

```diff
@@ -190,61 +274,97 @@ def doit(opts, args):
+
         myOutB = myOut.GetRasterBand(1)
         myOutNDV = myOutB.GetNoDataValue()
-        myOutType = gdal.GetDataTypeName(myOutB.DataType)
+        myOutType = myOutB.DataType
 
     else:
-        # remove existing file and regenerate
-        if os.path.isfile(opts.outF):
-            os.remove(opts.outF)
-        # create a new file
-        if opts.debug:
-            print("Generating output file %s" % (opts.outF))
+        if opts.outF:
+            # remove existing file and regenerate
+            if os.path.isfile(opts.outF):
+                os.remove(opts.outF)
+            # create a new file
+            if opts.debug:
+                print("Generating output file %s" % (opts.outF))
+        else:
+            opts.outF = ''
 
         # find data type to use
         if not opts.type:
             # use the largest type of the input files
-            myOutType = gdal.GetDataTypeName(max(myDataTypeNum))
+            myOutType = max(myDataTypeNum)
         else:
             myOutType = opts.type
+            if isinstance(myOutType, str):
+                myOutType = gdal.GetDataTypeByName(myOutType)
 
         # create file
         myOutDrv = gdal.GetDriverByName(opts.format)
         myOut = myOutDrv.Create(
             opts.outF, DimensionsCheck[0], DimensionsCheck[1], allBandsCount,
-            gdal.GetDataTypeByName(myOutType), opts.creation_options)
+            myOutType, opts.creation_options)
 
         # set output geo info based on first input layer
-        myOut.SetGeoTransform(myFiles[0].GetGeoTransform())
-        myOut.SetProjection(myFiles[0].GetProjection())
+        if not GeoTransformCheck:
+            GeoTransformCheck = myFiles[0].GetGeoTransform()
+        myOut.SetGeoTransform(GeoTransformCheck)
+        if not ProjectionCheck:
+            ProjectionCheck = myFiles[0].GetProjection()
+        myOut.SetProjection(ProjectionCheck)
 
         if opts.NoDataValue is not None:
-            myOutNDV = opts.NoDataValue
+            if opts.NoDataValue is ...:
+                # use opts.NoDataValue to indicate None datatype (opts.NoDataValue=None already indicates default value)
+                myOutNDV = None
+            else:
+                myOutNDV = opts.NoDataValue
         else:
             myOutNDV = DefaultNDVLookup[myOutType]
 
         for i in range(1, allBandsCount + 1):
             myOutB = myOut.GetRasterBand(i)
-            myOutB.SetNoDataValue(myOutNDV)
-            # write to band
-            myOutB = None
-
+            if myOutNDV is not None:
+                myOutB.SetNoDataValue(myOutNDV)
+            if opts.color_table:
+                # set color table and color interpretation
+                if isinstance(opts.color_table, (str, Path)):
+                    raise Exception('Error! reading a color table from a file is not supported yet')
```

Comment:
this contradicts the documentation of the option

Reply:
True.
As is you can only provide a color_palette object and not as a filename.
I actually implemented this feature in my codebase in ColorPalette.read()
https://github.com/talos-gis/gdalos/blob/master/src/gdalos/gdalos_color.py
The thing is that depends on other this fairly large class and I thought it's too big for this PR. I also thought that there might be already a built in function in GDAL to load a color_table from file, but I couldn't find it.
I would be happy to include this module in GDAL, unless there is already a better way to do so.
If I was to include the ColorPalette class in a different file, where should I put it so I can import it here?

Same issue regarding handling multiple GeoTransforms, my implementation depends on:
https://github.com/talos-gis/gdalos/blob/master/src/gdalos/rectangle.py

Reply:
I'd suggest to drop the color palette thing for now, so as to get the changes in this pull request reasonably small and reviewable.

Reply:
OK, I removed it from the parser for now. still available to use by calling the function with a color_table object